ui/src/app/shared/services/requests.ts

```diff
@@ -34,7 +34,17 @@ function apiRoot(): string {
     return toAbsURL('/api/v1');
 }
 
+const csrfHeaderName = 'X-Csrf-Token';
+let crsfToken: string;
+
 function initHandlers(req: agent.Request) {
+    req.set(csrfHeaderName, crsfToken || '');
+    req.on('response', res => {
+        const val = res.header['x-csrf-token'];
```

Comment:
NIT: I've seen some inconsistencies on the header casing for this in this PR:
* `X-Csrf-Token`
* `x-csrf-token`
* `X-CSRF-Token`

I don't think it affects anything, but since HTTP2 requires lowercase headers, might be worth converging on that standard.

Reply:
Makes sense, updated